nussl/datasets/hooks.py

```diff
@@ -417,6 +427,146 @@ def process_item(self, item):
                 "'mix', 'sources' as keys!")
         return output
 
+
+class SalientExcerptMixSourceFolder(OnTheFly):
+    """
+    Given a dataset in the expected format, this class identifies and
+    partitions the folder into a dataset containing segment_dur second audio
+    clips that contain a desired salient source. See MixSourceFolder for more
+    details on expected directory structure.
+
+    Args:
+        folder (str): Location that should be processed to produce the
+            list of files.
+        salient_src (str): The name of the source that will be used to identify
+         salient samples in the dataset. e.g. ('drums')
+        sample_rate (int, optional): the sampling rate for the audio files.
+        mix_folder (str, optional): Folder to look in for mixtures.
+         Defaults to 'mix'.
+        source_folders (list, optional): List of folders to look in for sources.
+            Path is defined relative to folder. If None, all folders other than
+            mix_folder are treated as the source folders. Defaults to None.
+        ext (list, optional): Audio extensions to look for in mix_folder.
+            Defaults to ['.wav', '.flac', '.mp3'].
+        threshold_db (float, optional): the minimum relative loudness of the
+         salient source measured in decibels relative to the rms of the salient
+         source. Defaults to -60.0.
+        segment_dur (float, optional): the duration of the desired audio clips
+         in seconds. Defaults to 4.0.
+        hop_ratio (float, optional): size of the hops to use when computing
+         the RMS. Defaults to 0.5.
+        verbose (bool, optional): suppress progress bar and logging for the
```

Comment:
This language is a bit ambiguous. Does "Suppress progress bar..." mean that if this arg is True it will suppress the progress bar? For bools, I like to be explicit in the docs: `If True, this flag will blah blah. Else, blah blah.`